### PR TITLE
feat: `x/blocksdk` module integration

### PR DESCRIPTION
**Pull request by aljo242**
- add `LaneFetcher` interface to `LanedMempool`
- update tests
- Update `LanedMempool.Registry()` to use `LaneFetcher` to on-the-fly build ordered lane list
- interchaintest

### CODE REVIEW

File: block/lane.go [hunk 1]
```diff
@@ -69,3 +72,15 @@ type Lane interface {
 	// Match determines if a transaction belongs to this lane.
 	Match(ctx sdk.Context, tx sdk.Tx) bool
 }
+
+// FindLane finds a Lanes from in an array of Lanes and returns it and its index if found.
+// Returns nil, 0 and false if not found.
+func FindLane(lanes []Lane, name string) (Lane, int, bool) {
```
Comment (keefertaylor) [inline block/lane.go:78]: Optional, but naming return types once you get to more than 2 fields in a tuple feels like a nice touch to me (but idk what the idiomatic thing to do is) 

```suggestion
func FindLane(lanes []Lane, name string) (lane Lane, index int, found bool) {
```

**Review by keefertaylor — APPROVED** [review]

File: block/mempool.go [hunk 2]
```diff
@@ -37,11 +45,17 @@ type (
...
 	}
 )
 
-// NewLanedMempool returns a new Block SDK LanedMempool. The laned mempool is
-// comprised of a registry of lanes. Each lane is responsible for selecting
+// NewLanedMempool returns a new Block SDK LanedMempool. The laned mempool comprises
+//
+//	a registry of lanes. Each lane is responsible for selecting
+//
 // transactions according to its own selection logic. The lanes are ordered
```
Comment (davidterpay) [inline block/mempool.go:59]: nit: format of docs

File: block/mempool.go [hunk 3]
```diff
@@ -63,8 +78,10 @@ func NewLanedMempool(logger log.Logger, mutex bool, lanes ...Lane) Mempool {
 
 	// Set the ignore list for each lane
 	if mutex {
-		registry := mempool.registry
-		for index, lane := range mempool.registry {
+		// perform full copy to prevent GC
+		registry := make([]Lane, len(mempool.registry))
+		copy(registry, mempool.registry)
```
Comment (davidterpay) [inline block/mempool.go:83]: kek

**Comment by davidterpay** [comment]
shippor shippin

**Review by davidterpay — APPROVED** [review]

**Pull request merged**